fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/DefaultOCFLObjectSession.java

```diff
@@ -449,6 +457,53 @@ private String commitUpdates(final CommitOption commitOption) {
         }
     }
 
+    private Consumer<OcflObjectUpdater> constructCommitUpdater(final boolean updatingObject) {
+        final DIGEST_ALGORITHM fcrepoDigestAlg = getObjectDigestAlgorithm();
+        final DigestAlgorithm ocflDigestAlg = translateFedoraDigestToOcfl(fcrepoDigestAlg);
+        final OcflOption[] ocflOptions;
+        if (updatingObject) {
+            ocflOptions = new OcflOption[] { MOVE_SOURCE, OVERWRITE };
+        } else {
+            ocflOptions = new OcflOption[] { MOVE_SOURCE };
+        }
+
+
+        return updater -> {
+            if (updatingObject) {
+                deletePaths.forEach(updater::removeFile);
+            }
+            if (isStagingEmpty()) {
+                return;
+            }
+
+            try {
+                Files.walk(stagingPath)
```

Comment:
I don't think you need to walk the staged files, unless you want to. You should just be able to add the entire staging directory, and then iterate over your map of digests, calling `addFileFixity` for each.

Reply:
Good catch, I was still in the mindset of needing to walk files, but there's really no need at this point. I've switched to your suggestion